In [ ]:
from google.colab import drive

In [ ]:
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# IMDb Movies Revenue & Profitability Analysis using Python

## Objective
Analyzed IMDb movie dataset to identify revenue growth patterns,
profitability drivers, language dominance, and release trends
using Python and Pandas.



In [ ]:
import pandas as pd
import numpy as np

In [ ]:

data = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/imdb_data.csv')
data

id                              belongs_to_collection    budget  \
0        1  [{'id': 313576, 'name': 'Hot Tub Time Machine ...  14000000   
1        2  [{'id': 107674, 'name': 'The Princess Diaries ...  40000000   
2        3                                                NaN   3300000   
3        4                                                NaN   1200000   
4        5                                                NaN         0   
...    ...                                                ...       ...   
2995  2996                                                NaN         0   
2996  2997                                                NaN         0   
2997  2998                                                NaN  65000000   
2998  2999                                                NaN  42000000   
2999  3000                                                NaN  35000000   

                                                 genres  \
0                        [{'id': 35, 'name': 'Comedy'}]   
1     [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
2                         [{'id': 18, 'name': 'Drama'}]   
3     [{'id': 53, 'name': 'Thriller'}, {'id': 18, 'n...   
4     [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...   
...                                                 ...   
2995  [{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...   
2996  [{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...   
2997  [{'id': 80, 'name': 'Crime'}, {'id': 28, 'name...   
2998  [{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...   
2999  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...   

                               homepage    imdb_id original_language  \
0                                   NaN  tt2637294                en   
1                                   NaN  tt0368933                en   
2     http://sonyclassics.com/whiplash/  tt2582802                en   
3            http://kahaanithefilm.com/  tt1821480                hi   
4                                   NaN  tt1380152                ko   
...                                 ...        ...               ...   
2995                                NaN  tt0109403                en   
2996                                NaN  tt2364975                sv   
2997                                NaN  tt0116908                en   
2998     http://www.alongcamepolly.com/  tt0343135                en   
2999   http://www.abductionthefilm.com/  tt1600195                en   

                                original_title  \
0                       Hot Tub Time Machine 2   
1     The Princess Diaries 2: Royal Engagement   
2                                     Whiplash   
3                                      Kahaani   
4                                         마린보이   
...                                        ...   
2995                                   Chasers   
2996                               Vi är bäst!   
2997                   The Long Kiss Goodnight   
2998                          Along Came Polly   
2999                                 Abduction   

                                               overview  popularity  ...  \
0     When Lou, who has become the "father of the In...    6.575393  ...   
1     Mia Thermopolis is now a college graduate and ...    8.248895  ...   
2     Under the direction of a ruthless instructor, ...   64.299990  ...   
3     Vidya Bagchi (Vidya Balan) arrives in Kolkata ...    3.174936  ...   
4     Marine Boy is the story of a former national s...    1.148070  ...   
...                                                 ...         ...  ...   
2995  Military men Rock Reilly and Eddie Devane are ...    9.853270  ...   
2996  Three girls in 1980s Stockholm decide to form ...    3.727996  ...   
2997  Samantha Caine, suburban homemaker, is the ide...   14.482345  ...   
2998  Reuben Feffer is a guy who's spent his entire ...   15.725542  ...   
2999  A young man sets out to uncover the truth abou...   10.512109  ...   

     release_date r

#Initial Exploration

In [ ]:
data.shape

(3000, 23)

Dataset: ~3000 movies | 23 features including revenue, budget, runtime, language, genres, popularity, and release dates.

In [ ]:
data.head()

id                              belongs_to_collection    budget  \
0   1  [{'id': 313576, 'name': 'Hot Tub Time Machine ...  14000000   
1   2  [{'id': 107674, 'name': 'The Princess Diaries ...  40000000   
2   3                                                NaN   3300000   
3   4                                                NaN   1200000   
4   5                                                NaN         0   

                                              genres  \
0                     [{'id': 35, 'name': 'Comedy'}]   
1  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
2                      [{'id': 18, 'name': 'Drama'}]   
3  [{'id': 53, 'name': 'Thriller'}, {'id': 18, 'n...   
4  [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...   

                            homepage    imdb_id original_language  \
0                                NaN  tt2637294                EN   
1                                NaN  tt0368933                EN   
2  http://sonyclassics.com/whiplash/  tt2582802                EN   
3         http://kahaanithefilm.com/  tt1821480                HI   
4                                NaN  tt1380152                KO   

                             original_title  \
0                    Hot Tub Time Machine 2   
1  The Princess Diaries 2: Royal Engagement   
2                                  Whiplash   
3                                   Kahaani   
4                                      마린보이   

                                            overview  popularity  ...  \
0  When Lou, who has become the "father of the In...    6.575393  ...   
1  Mia Thermopolis is now a college graduate and ...    8.248895  ...   
2  Under the direction of a ruthless instructor, ...   64.299990  ...   
3  Vidya Bagchi (Vidya Balan) arrives in Kolkata ...    3.174936  ...   
4  Marine Boy is the story of a former national s...    1.148070  ...   

  release_date runtime                                   spoken_languages  \
0      2/20/15    93.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1       8/6/04   113.0           [{'iso_639_1': 'en', 'name': 'English'}]   
2     10/10/14   105.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3       3/9/12   122.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
4       2/5/09   118.0           [{'iso_639_1': 'ko', 'name': '한국어/조선말'}]   

     status                                            tagline  \
0  Released  The Laws of Space and Time are About to be Vio...   
1  Released  It can take a lifetime to find true love; she'...   
2  Released    The road to greatness can take you to the edge.   
3  Released                                                NaN   
4  Released                                                NaN   

                                      title  \
0                    Hot Tub Time Machine 2   
1  The Princess Diaries 2: Royal Engagement   
2                                  Whiplash   
3                                   Kahaani   
4                                Marine Boy   

                                            Keywords  \
0  [{'id': 4379, 'name': 'time travel'}, {'id': 9...   
1  [{'id': 2505, 'name': 'coronation'}, {'id': 42...   
2  [{'id': 1416, 'name': 'jazz'}, {'id': 1523, 'n...   
3  [{'id': 10092, 'name': 'mystery'}, {'id': 1054...   
4                                                NaN   

                                                cast  \
0  [{'cast_id': 4, 'character': 'Lou', 'credit_id...   
1  [{'cast_id': 1, 'character': 'Mia Thermopolis'...   
2  [{'cast_id': 5, 'character': 'Andrew Neimann',...   
3  [{'cast_id': 1, 'character': 'Vidya Bagchi', '...   
4  [{'cast_id': 3, 'character': 'Chun-soo', 'cred...   

                                                crew   revenue  
0  [{'credit_id': '59ac067c92514107af02c8c8', 'de...  12314651  
1  [{'credit_id': '52fe43fe9251416c7502563d', 'de...  95149435  
2  [{'credit_id': '54d5356ec3a3683ba0000039', 'de...  13092000  
3  [{'credit_id': '52fe48779251

In [ ]:
data.tail()

Id                              belongs_to_collection     budget  \
2904  2905  [{'id': 424202, 'name': 'Trainspotting Collect...    4000000   
2927  2928  [{'id': 645, 'name': 'James Bond Collection', ...  140000000   
2938  2939  [{'id': 135416, 'name': 'Prometheus Collection...  130000000   
2968  2969  [{'id': 97307, 'name': 'BloodRayne Collection'...   25000000   
2984  2985  [{'id': 221111, 'name': 'S.W.A.T. Collection',...   80000000   

                                                 genres  \
2904  [{'id': 18, 'name': 'Drama'}, {'id': 80, 'name...   
2927  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   
2938  [{'id': 878, 'name': 'Science Fiction'}, {'id'...   
2968  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
2984  [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...   

                                               homepage    imdb_id  \
2904        http://www.miramax.com/movie/trainspotting/  tt0117951   
2927  http://www.mgm.com/view/movie/232/Die-Another-...  tt0246460   
2938                  http://www.projectprometheus.com/  tt1446714   
2968  http://www.bloodrayne-themovie.com/main/index....  tt0383222   
2984        http://www.sonypictures.com/homevideo/swat/  tt0257076   

     original_language   original_title  \
2904                EN    TRAINSPOTTING   
2927                EN  DIE ANOTHER DAY   
2938                EN       PROMETHEUS   
2968                EN       BLOODRAYNE   
2984                EN         S.W.A.T.   

                                               overview  popularity  ...  \
2904  Renton, deeply immersed in the Edinburgh drug ...   19.348466  ...   
2927  Bond takes on a North Korean leader who underg...   12.996474  ...   
2938  A team of explorers discover a clue to the ori...   16.624854  ...   
2968  In eighteenth century Romania, Rayne, a dhampi...    6.514132  ...   
2984  Hondo Harrelson recruits Jim Street to join an...   13.334952  ...   

     month                                         month_Name day  \
2904     2  <bound method PandasDelegate._add_delegate_acc...  23   
2927    11  <bound method PandasDelegate._add_delegate_acc...  17   
2938     5  <bound method PandasDelegate._add_delegate_acc...  30   
2968    10  <bound method PandasDelegate._add_delegate_acc...  22   
2984     8  <bound method PandasDelegate._add_delegate_acc...   8   

     capitalized_title     profit new_profit popularity_new profit_value  \
2904     TRAINSPOTTING   12491080   12491080          19.35     positive   
2927   DIE ANOTHER DAY  291971116  291971116          13.00     positive   
2938        PROMETHEUS  273170142  273170142          16.62     positive   
2968        BLOODRAYNE  -22594580  -22594580           6.51     negative   
2984          S.W.A.T.   36643346   36643346          13.33     positive   

                            genre_names release_year  
2904                        Drama,Crime         1996  
2927          Adventure,Action,Thriller         2002  
2938  Science Fiction,Adventure,Mystery         2012  
2968    Action,Adventure,Fantasy,Horror         2005  
2984              Action,Thriller,Crime         2003  

[5 rows x 37 columns]

In [ ]:
data.columns

Index(['id', 'belongs_to_collection', 'budget', 'genres', 'homepage',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'runtime', 'spoken_languages',
       'status', 'tagline', 'title', 'Keywords', 'cast', 'crew', 'revenue'],
      dtype='object')

In [ ]:
data.dtypes

id                         int64
belongs_to_collection     object
budget                     int64
genres                    object
homepage                  object
imdb_id                   object
original_language         object
original_title            object
overview                  object
popularity               float64
poster_path               object
production_companies      object
production_countries      object
release_date              object
runtime                  float64
spoken_languages          object
status                    object
tagline                   object
title                     object
Keywords                  object
cast                      object
crew                      object
revenue                    int64
dtype: object

In [ ]:
num_data = data.select_dtypes(include =['int64','float64'])

In [ ]:
num_data.columns

Index(['id', 'budget', 'popularity', 'runtime', 'revenue'], dtype='object')

In [ ]:
cat_data = data.select_dtypes(include = ['object'])
cat_data.columns

Index(['belongs_to_collection', 'genres', 'homepage', 'imdb_id',
       'original_language', 'original_title', 'overview', 'poster_path',
       'production_companies', 'production_countries', 'release_date',
       'spoken_languages', 'status', 'tagline', 'title', 'Keywords', 'cast',
       'crew'],
      dtype='object')

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 23 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   id                     3000 non-null   int64  
 1   belongs_to_collection  604 non-null    object 
 2   budget                 3000 non-null   int64  
 3   genres                 2993 non-null   object 
 4   homepage               946 non-null    object 
 5   imdb_id                3000 non-null   object 
 6   original_language      3000 non-null   object 
 7   original_title         3000 non-null   object 
 8   overview               2992 non-null   object 
 9   popularity             3000 non-null   float64
 10  poster_path            2999 non-null   object 
 11  production_companies   2844 non-null   object 
 12  production_countries   2945 non-null   object 
 13  release_date           3000 non-null   object 
 14  runtime                2998 non-null   float64
 15  spok

In [ ]:
data.describe()

id        budget   popularity      runtime       revenue
count  3000.000000  3.000000e+03  3000.000000  2998.000000  3.000000e+03
mean   1500.500000  2.253133e+07     8.463274   107.856571  6.672585e+07
std     866.169729  3.702609e+07    12.104000    22.086434  1.375323e+08
min       1.000000  0.000000e+00     0.000001     0.000000  1.000000e+00
25%     750.750000  0.000000e+00     4.018053    94.000000  2.379808e+06
50%    1500.500000  8.000000e+06     7.374861   104.000000  1.680707e+07
75%    2250.250000  2.900000e+07    10.890983   118.000000  6.891920e+07
max    3000.000000  3.800000e+08   294.337037   338.000000  1.519558e+09

In [ ]:
data.describe(include='object')

belongs_to_collection  \
count                                                 604   
unique                                                422   
top     [{'id': 645, 'name': 'James Bond Collection', ...   
freq                                                   16   

                               genres                           homepage  \
count                            2993                                946   
unique                            872                                941   
top     [{'id': 18, 'name': 'Drama'}]  http://www.transformersmovie.com/   
freq                              266                                  4   

          imdb_id original_language original_title  \
count        3000              3000           3000   
unique       3000                36           2975   
top     tt1600195                EN  Casino Royale   
freq            1              2575              2   

                                                 overview  \
count                                                2992   
unique                                               2992   
top     Anna (Marceau) is a wife and mother who has an...   
freq                                                    1   

                             poster_path  \
count                               2999   
unique                              2999   
top     /cUT6NQP5LAJpmUoStGtXmvNt4zA.jpg   
freq                                   1   

                             production_companies  \
count                                        2844   
unique                                       2383   
top     [{'name': 'Paramount Pictures', 'id': 4}]   
freq                                           51   

                                     production_countries release_date  \
count                                                2945         3000   
unique                                                321         2398   
top     [{'iso_3166_1': 'US', 'name': 'United States o...      9/10/15   
freq                                                 1752            5   

                                spoken_languages    status  \
count                                       2980      3000   
unique                                       401         2   
top     [{'iso_639_1': 'en', 'name': 'English'}]  Released   
freq                                        1817      2996   

                       tagline          title  \
count                     2403           3000   
unique                    2400           2969   
top     Based on a true story.  Loose Cannons   
freq                         3              2   

                                           Keywords  cast  \
count                                          2724  2987   
unique                                         2648  2975   
top     [{'id': 10183, 'name': 'independent film'}]    []   
freq                                             27    13   

                                                     crew  
count                                                2984  
unique                                               2984  
top     [{'credit_id': '5391990d0e0a260fb5001629', 'de...  
freq                                                    1

The dataset contains both numerical and categorical features,
including revenue, budget, runtime, language, and genres.


#Data Cleaning

In [ ]:
data.isnull()

id  belongs_to_collection  budget  genres  homepage  imdb_id  \
0     False                  False   False   False      True    False   
1     False                  False   False   False      True    False   
2     False                   True   False   False     False    False   
3     False                   True   False   False     False    False   
4     False                   True   False   False      True    False   
...     ...                    ...     ...     ...       ...      ...   
2995  False                   True   False   False      True    False   
2996  False                   True   False   False      True    False   
2997  False                   True   False   False      True    False   
2998  False                   True   False   False     False    False   
2999  False                   True   False   False     False    False   

      original_language  original_title  overview  popularity  ...  \
0                 False           False     False       False  ...   
1                 False           False     False       False  ...   
2                 False           False     False       False  ...   
3                 False           False     False       False  ...   
4                 False           False     False       False  ...   
...                 ...             ...       ...         ...  ...   
2995              False           False     False       False  ...   
2996              False           False     False       False  ...   
2997              False           False     False       False  ...   
2998              False           False     False       False  ...   
2999              False           False     False       False  ...   

      release_date  runtime  spoken_languages  status  tagline  title  \
0            False    False             False   False    False  False   
1            False    False             False   False    False  False   
2            False    False             False   False    False  False   
3            False    False             False   False     True  False   
4            False    False             False   False     True  False   
...            ...      ...               ...     ...      ...    ...   
2995         False    False             False   False    False  False   
2996         False    False             False   False     True  False   
2997         False    False             False   False    False  False   
2998         False    False             False   False    False  False   
2999         False    False             False   False    False  False   

      Keywords   cast   crew  revenue  
0        False  False  False    False  
1        False  False  False    False  
2        False  False  False    False  
3        False  False  False    False  
4         True  False  False    False  
...        ...    ...    ...      ...  
2995     False  False  False    False  
2996     False  False  False    False  
2997     False  False  False    False  
2998     False  False  False    False  
2999     False  False  False    False  

[3000 rows x 23 columns]

In [ ]:
data.isnull().sum()

id                          0
belongs_to_collection    2396
budget                      0
genres                      7
homepage                 2054
imdb_id                     0
original_language           0
original_title              0
overview                    8
popularity                  0
poster_path                 1
production_companies      156
production_countries       55
release_date                0
runtime                     2
spoken_languages           20
status                      0
tagline                   597
title                       0
Keywords                  276
cast                       13
crew                       16
revenue                     0
dtype: int64

In [ ]:
data.dropna(inplace= True)

In [ ]:
data.isnull().sum()

id                       0
belongs_to_collection    0
budget                   0
genres                   0
homepage                 0
imdb_id                  0
original_language        0
original_title           0
overview                 0
popularity               0
poster_path              0
production_companies     0
production_countries     0
release_date             0
runtime                  0
spoken_languages         0
status                   0
tagline                  0
title                    0
Keywords                 0
cast                     0
crew                     0
revenue                  0
dtype: int64

In [ ]:
data.duplicated().sum()

np.int64(0)

In [ ]:
data.drop_duplicates(inplace = True)

### Data Cleaning Insight
- Missing values and duplicate records were removed
- This ensures accuracy in revenue and profit analysis


In [ ]:
data['budget']

22              0
40              0
46       80000000
60      150000000
69       85000000
          ...    
2904      4000000
2927    140000000
2938    130000000
2968     25000000
2984     80000000
Name: budget, Length: 203, dtype: int64

In [ ]:
data[['original_language','original_title','overview','popularity']]

original_language                         original_title  \
22                  EN                                  V/H/S   
40                  EN                        Cocaine Cowboys   
46                  EN                  Ice Age: The Meltdown   
60                  EN                          Casino Royale   
69                  EN  The Fast and the Furious: Tokyo Drift   
...                ...                                    ...   
2904                EN                          Trainspotting   
2927                EN                        Die Another Day   
2938                EN                             Prometheus   
2968                EN                             BloodRayne   
2984                EN                               S.W.A.T.   

                                               overview  popularity  
22    When a group of misfits is hired by an unknown...    7.820787  
40    In the 1980s, ruthless Colombian cocaine baron...    2.941626  
46    Diego, Manny and Sid return in this sequel to ...   16.646029  
60    Le Chiffre, a banker to the world's terrorists...   23.065078  
69    In order to avoid a jail sentence, Sean Boswel...    2.238808  
...                                                 ...         ...  
2904  Renton, deeply immersed in the Edinburgh drug ...   19.348466  
2927  Bond takes on a North Korean leader who underg...   12.996474  
2938  A team of explorers discover a clue to the ori...   16.624854  
2968  In eighteenth century Romania, Rayne, a dhampi...    6.514132  
2984  Hondo Harrelson recruits Jim Street to join an...   13.334952  

[203 rows x 4 columns]

In [ ]:
data.sort_values('revenue',ascending=False)

id                              belongs_to_collection     budget  \
1126  1127  [{'id': 86311, 'name': 'The Avengers Collectio...  220000000   
1761  1762  [{'id': 9485, 'name': 'The Fast and the Furiou...  190000000   
2770  2771  [{'id': 86311, 'name': 'The Avengers Collectio...  280000000   
2322  2323  [{'id': 8650, 'name': 'Transformers Collection...  195000000   
2135  2136  [{'id': 295, 'name': 'Pirates of the Caribbean...  380000000   
...    ...                                                ...        ...   
514    515  [{'id': 206559, 'name': 'StreetDance Collectio...          0   
2276  2277  [{'id': 286038, 'name': 'Dead Snow Collection'...          0   
1499  1500  [{'id': 473843, 'name': 'Stake Land Collection...     650000   
1563  1564  [{'id': 87186, 'name': 'The Boondock Saints Co...    6000000   
665    666  [{'id': 86668, 'name': 'Elektra Luxx Collectio...          0   

                                                 genres  \
1126  [{'id': 878, 'name': 'Science Fiction'}, {'id'...   
1761                     [{'id': 28, 'name': 'Action'}]   
2770  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
2322  [{'id': 28, 'name': 'Action'}, {'id': 878, 'na...   
2135  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   
...                                                 ...   
514   [{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...   
2276  [{'id': 27, 'name': 'Horror'}, {'id': 28, 'nam...   
1499  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...   
1563  [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...   
665   [{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...   

                                               homepage    imdb_id  \
1126                  http://marvel.com/avengers_movie/  tt0848228   
1761                           http://www.furious7.com/  tt2820852   
2770  http://marvel.com/movies/movie/193/avengers_ag...  tt2395427   
2322                  http://www.transformersmovie.com/  tt1399103   
2135  http://disney.go.com/pirates/index-on-stranger...  tt1298650   
...                                                 ...        ...   
514                  http://www.highstrungthemovie.com/  tt3451230   
2276                           http://www.deadsnow.com/  tt2832470   
1499                   http://www.stakelandthefilm.com/  tt1464580   
1563                   http://www.theboondocksaints.com  tt0144117   
665                   http://www.elektraluxx-movie.com/  tt1340773   

     original_language                               original_title  \
1126                EN                                 The Avengers   
1761                EN                                    Furious 7   
2770                EN                      Avengers: Age of Ultron   
2322                EN               Transformers: Dark of the Moon   
2135                EN  Pirates of the Caribbean: On Stranger Tides   
...                ...                                          ...   
514                 EN                                  High Strung   
2276                NO                                    Død Snø 2   
1499                EN                                   Stake Land   
1563                EN                          The Boondock Saints   
665                 EN                                 Elektra Luxx   

                                               overview  popularity  ...  \
1126  When an unexpected enemy emerges and threatens...   89.887648  ...   
1761  Deckard Shaw seeks revenge against Dominic Tor...   27.275687  ...   
2770  When Tony Stark tries to jumpstart a dormant p...   37.379420  ...   
2322  Sam Witwicky takes his first tenuous steps int...    4.503505  ...   
2135  Captain Jack Sparrow crosses paths with a woma...   27.887720  ...   
...                                                 ...         ...  ...   
514   When a hip hop violinist busking in the New Yo...    6.625607  ...   
2276  The gruesome Nazi Zombies are back to finish t...   10.287127  ...   
1499  Ma

#Feature Engineering

In [ ]:
data.rename(columns={'revenue':'Revenue','id':'Id','production_companies':'Companies','production_countries':'Countries'},inplace=True)

In [ ]:
data.loc[:,'Half_runtime'] = data.loc[:,'runtime']/2

#Date Processing

In [ ]:
data.loc[:,'new_released_date']=pd.to_datetime(data.loc[:,'release_date'])

/tmp/ipython-input-455865037.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  data.loc[:,'new_released_date']=pd.to_datetime(data.loc[:,'release_date'])


In [ ]:
data.loc[:,'year'] = data.loc[:,'new_released_date'].dt.year

In [ ]:
data.loc[:,'month']=data.loc[:,'new_released_date'].dt.month

In [ ]:
data.loc[:,'month_Name']=data.loc[:,'new_released_date'].dt.month_name

In [ ]:
data.loc[:,'day']=data.loc[:,'new_released_date'].dt.day

Movies released after 2010 show higher cumulative revenue,
indicating growth in the global film industry.


#Language Standardization

In [ ]:
data['original_language'].value_counts()

original_language
EN    179
JA      5
ES      3
NO      2
FR      1
KO      1
ZH      1
SV      1
CN      1
RU      1
ID      1
Name: count, dtype: int64

In [ ]:
data['original_language'].unique()

array(['en', 'ja', 'es', 'fr', 'no', 'ko', 'zh', 'sv', 'cn', 'ru', 'id'],
      dtype=object)

In [ ]:
data.loc[data.loc[:,"year"]>2014]

Id                              belongs_to_collection     budget  \
211    212  [{'id': 2366, 'name': 'The Jaws Collection', '...    7000000   
294    295  [{'id': 131635, 'name': 'The Hunger Games Coll...  160000000   
314    315  [{'id': 87118, 'name': 'Cars Collection', 'pos...  175000000   
351    352  [{'id': 645, 'name': 'James Bond Collection', ...    7200000   
514    515  [{'id': 206559, 'name': 'StreetDance Collectio...          0   
548    549  [{'id': 426103, 'name': 'Ouija Collection', 'p...    9000000   
699    700  [{'id': 376650, 'name': 'Ride Along Collection...   40000000   
809    810  [{'id': 115776, 'name': 'Robert Langdon Collec...   75000000   
818    819  [{'id': 167613, 'name': 'Alvin and the Chipmun...          0   
837    838  [{'id': 8945, 'name': 'Mad Max Collection', 'p...  150000000   
1038  1039  [{'id': 401562, 'name': 'Teenage Mutant Ninja ...  135000000   
1093  1094  [{'id': 230, 'name': 'The Godfather Collection...    6000000   
1124  1125  [{'id': 8354, 'name': 'Ice Age Collection', 'p...  105000000   
1514  1515  [{'id': 357173, 'name': 'Sinister Collection',...   10000000   
1630  1631  [{'id': 8650, 'name': 'Transformers Collection...  260000000   
1643  1644  [{'id': 295130, 'name': 'The Maze Runner Colle...   61000000   
1715  1716  [{'id': 173710, 'name': 'Planet of the Apes (R...  152000000   
1761  1762  [{'id': 9485, 'name': 'The Fast and the Furiou...  190000000   
2030  2031  [{'id': 281644, 'name': 'Rocky Horror Collecti...    1200000   
2126  2127  [{'id': 284433, 'name': 'Guardians of the Gala...  200000000   
2274  2275  [{'id': 386534, 'name': '... Has Fallen Collec...   60000000   
2326  2327  [{'id': 448150, 'name': 'Deadpool Collection',...   58000000   
2333  2334  [{'id': 645, 'name': 'James Bond Collection', ...    9500000   
2349  2350  [{'id': 12263, 'name': 'The Exorcist Collectio...    8000000   
2391  2392  [{'id': 645, 'name': 'James Bond Collection', ...    5500000   
2431  2432  [{'id': 313086, 'name': 'The Conjuring Collect...   40000000   
2562  2563  [{'id': 137697, 'name': 'Finding Nemo Collecti...  200000000   
2737  2738  [{'id': 645, 'name': 'James Bond Collection', ...  245000000   
2770  2771  [{'id': 86311, 'name': 'The Avengers Collectio...  280000000   
2890  2891  [{'id': 14563, 'name': 'The Ring Collection', ...   25000000   

                                                 genres  \
211   [{'id': 27, 'name': 'Horror'}, {'id': 53, 'nam...   
294   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
314   [{'id': 10751, 'name': 'Family'}, {'id': 35, '...   
351   [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   
514   [{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...   
548   [{'id': 27, 'name': 'Horror'}, {'id': 53, 'nam...   
699   [{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...   
809                    [{'id': 53, 'name': 'Thriller'}]   
818   [{'id': 12, 'name': 'Adventure'}, {'id': 16, '...   
837   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
1038  [{'id': 14, 'name': 'Fantasy'}, {'id': 28, 'na...   
1093  [{'id': 18, 'name': 'Drama'}, {'id': 80, 'name...   
1124  [{'id': 12, 'name': 'Adventure'}, {'id': 16, '...   
1514                     [{'id': 27, 'name': 'Horror'}]   
1630  [{'id': 28, 'name': 'Action'}, {'id': 878, 'na...   
1643                     [{'id': 28, 'name': 'Action'}]   
1715  [{'id': 18, 'name': 'Drama'}, {'id': 878, 'nam...   
1761                     [{'id': 28, 'name': 'Action'}]   
2030  [{'id': 35, 'name': 'Comedy'}, {'id': 27, 'nam...   
2126  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
2274  [{'id': 28, 'name': 'Action'}, {'id': 80, 'nam...   
2326  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
2333  [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...   
2349  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...   
2391  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   
2431                     [{'id': 27, 'name': 'Horror'}]   
2562  [{'id': 12, 'name': 'Adventure'}, {'id': 16, '...   

English-language movies dominate total revenue,
highlighting wider global reach.


In [ ]:
data.loc[(data.loc[:,'year']=='2012') | (data.loc[:,'year']=='2013')]

Empty DataFrame
Columns: [Id, belongs_to_collection, budget, genres, homepage, imdb_id, original_language, original_title, overview, popularity, poster_path, Companies, Countries, release_date, runtime, spoken_languages, status, tagline, title, Keywords, cast, crew, Revenue, Profit, Half_runtime, new_released_date, year, month, month_Name, day]
Index: []

[0 rows x 30 columns]

In [ ]:
def to_uppercase(column):
  return column.upper()

data.loc[:,"capitalized_title"] = data.loc[:,"title"].apply(to_uppercase)


In [ ]:
data.loc[:,'original_title']= data.loc[:,'original_title'].apply(to_uppercase)

In [ ]:
def to_lowercase(value):
  if isinstance(value,str):
    return value.lower()
  return value



In [ ]:
data.loc[:,'tagline'] = data.loc[:,'tagline'].apply(to_lowercase)

### Profit Analysis
Profit was calculated as Revenue minus Budget for each movie. This metric helps
identify which movies were commercially successful, beyond just high revenue.

In [ ]:
def calculate_profit(row):
  return row['Revenue'] - row['budget']

data.loc[:,'profit'] = data.apply(calculate_profit,axis=1)

In [ ]:
def revised_profit(revenue,budget):
  return revenue-budget

data['new_profit'] = data.apply(lambda x: revised_profit(x['Revenue'], x['budget']), axis=1)

In [ ]:
data.loc[:,'profit_value'] = data.loc[:,'profit'].apply(lambda x: 'positive' if x>0 else 'negative')

In [ ]:
data.loc[:,'original_language'] = data.loc[:,'original_language'].apply(lambda x: x.upper())

In [ ]:
data.loc[:,'popularity_new'] = data.loc[:,'popularity'].apply(lambda x: round(x,2))

In [ ]:
import ast

def func(value, key):
    try:
        j = ast.literal_eval(value)
        return ",".join([i[key] for i in j])
    except:
        return np.nan

Genres are stored as JSON-like strings and were parsed to extract genre names.

In [ ]:
data.loc[:,'genre_names'] = data.loc[:,'genres'].apply(func, key='name')

In [ ]:
data.loc[:,'release_date'] = pd.to_datetime(data.loc[:,'release_date'])

/tmp/ipython-input-1722189326.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  data.loc[:,'release_date'] = pd.to_datetime(data.loc[:,'release_date'])


In [ ]:
data.loc[:,'release_year'] = pd.to_datetime(data.loc[:,'release_date']).dt.year

#GroupBy Insights

In [ ]:
data = data[data['release_year']<=2017].copy()
data.groupby('release_year')['Revenue'].sum().sort_index(ascending = False)

release_year
2017    2271517194
2016    3418418977
2015    5422508761
2014    4066477009
2013    3408564261
2012    5192265377
2011    4055518627
2010    3124718173
2009    5044521315
2008    3116785148
2007    3103397036
2006    2347970243
2005    2042091705
2004    1573792676
2003     983251296
2002    2659232022
2001    1150797240
2000     127879550
1999    1191812141
1997     333011068
1996      84605061
1995     446409952
1994     801164564
1993     920100000
1992     352927224
1990     381293645
1989    1002823860
1988       7282851
1987     405140171
1985     152427960
1984     411371200
1981     245557502
1979     104931801
1977     185438673
Name: Revenue, dtype: int64

### Revenue Trend by Release Year

Total movie revenue shows a clear growth pattern from the late 1990s,
with a strong surge between 2009 and 2015. The highest total revenue
was observed in 2015, followed by consistently high revenues until 2016.
A noticeable decline is seen in 2017, which may be influenced by fewer
movies or incomplete data for that year.

In [ ]:
data.groupby('original_language')['Revenue'].sum().sort_values(ascending=False)


original_language
EN    59422701124
JA      377352903
ZH      121059225
SV       66995253
ES       49107680
RU       38862717
KO       23803308
CN       21888598
FR        9584131
ID        2627209
NO        2022135
Name: Revenue, dtype: int64

### Revenue by Original Language

**English**-language movies overwhelmingly dominate total revenue,
contributing the vast majority of box-office earnings. Other languages
such as Japanese (JA), Chinese (ZH), and Swedish (SV) generate
significantly lower revenue in comparison. This highlights the strong
global reach and commercial dominance of English-language films.

In [ ]:
data.groupby('release_year')[['Revenue','budget']].sum().sort_index(ascending = False)

Revenue      budget
release_year                        
2017          2271517194   812000000
2016          3418418977   722000000
2015          5422508761  1096000000
2014          4066477009  1059000000
2013          3408564261  1003000000
2012          5192265377  1244500000
2011          4055518627  1147000000
2010          3124718173   732487625
2009          5044521315   967400000
2008          3116785148   773715578
2007          3103397036   956515000
2006          2347970243   579580000
2005          2042091705   553000000
2004          1573792676   442700000
2003           983251296   264000000
2002          2659232022   491000000
2001          1150797240   273000000
2000           127879550    99000000
1999          1191812141   321060000
1997           333011068   110000000
1996            84605061    73000000
1995           446409952    76500000
1994           801164564    80027000
1993           920100000    63000000
1992           352927224    49000000
1990           381293645    94000000
1989          1002823860   130000000
1988             7282851     3000000
1987           405140171    46000000
1985           152427960    30000000
1984           411371200    34400000
1981           245557502    34000000
1979           104931801    11000000
1977           185438673    14000000

### Revenue vs Budget Trend by Release Year

Both movie budgets and revenues show a steady increase over time,
indicating higher investment and scale in the film industry. From the
early 2000s onward, revenue growth generally outpaces budget growth,
suggesting improved profitability during peak years such as 2009, 2012,
and 2015. Although budgets continue to rise, revenue fluctuations in
later years indicate that higher investment does not always guarantee
proportionally higher returns.

In [ ]:
data.groupby('release_year')['runtime'].max().reset_index()

release_year  runtime
0           1977    125.0
1           1979    117.0
2           1981    127.0
3           1984    118.0
4           1985    131.0
5           1987    130.0
6           1988     97.0
7           1989    133.0
8           1990    162.0
9           1992    127.0
10          1993    127.0
11          1994     92.0
12          1995    130.0
13          1996    108.0
14          1997    119.0
15          1999    128.0
16          2000     93.0
17          2001    178.0
18          2002    179.0
19          2003    143.0
20          2004    132.0
21          2005    140.0
22          2006    144.0
23          2007    169.0
24          2008    152.0
25          2009    153.0
26          2010    133.0
27          2011    154.0
28          2012    169.0
29          2013    161.0
30          2014    150.0
31          2015    148.0
32          2016    134.0
33          2017    149.0

### Maximum Movie Runtime by Release Year

The maximum runtime of movies shows a gradual increase over time,
indicating a trend toward longer feature films in later years. Earlier
years (1970s–1980s) generally had maximum runtimes around 120–130 minutes,
whereas from the 2000s onward, several years feature movies exceeding
150 minutes. This suggests growing audience acceptance of longer films,
often associated with epic storytelling, franchises, or high-budget
productions.